docs/guides/src/main/operator/advanced-configuration.adoc

```diff
@@ -0,0 +1,113 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Advanced configuration"
+summary="How to tune advanced aspects of the Keycloak CR">
+
+== Advanced Configuration
+In this guide we will show how advanced concepts and options provided by Custom Resources (CR).
+
+=== Server Configuration details
+
+The `serverConfiguration` field of the Keycloak CR allow to pass to Keycloak any available configuration in the form of key-value pairs.
+For all the available configuration options refer to <@links.server id="all-config"/>.
+
+The values can be expressed as plain text strings or Kubernetes Secret references.
+e.g:
+
+[source,yaml]
+----
+apiVersion: keycloak.org/v2alpha1
+kind: Keycloak
+metadata:
+  name: example-kc
+spec:
+  ...
+  serverConfiguration:
+    - name: db
+      value: postgres # plain text value
+    - name: db-url-host
+      value: postgres-db # plain text value
+    - name: db-username
+      secret: # Secret reference
+        name: keycloak-db-secret # name of the Secret
+        key: username # name of the Key in the Secret
+    - name: db-password
+      secret: # secret reference
+        name: keycloak-db-secret # name of the Secret
+        key: password # name of the Key in the Secret
+----
+
+=== Secret References
+
+A Secret Reference can be either a value in `serverConfiguration` or the `tlsSecret`.
+
+When specifying a Secret Reference you have to make sure that a Secret containing the referenced keys is present in the same namespace as the CR referencing it.
+Along with the Keycloak Server Deployment the operator will add special labels to the referenced Secrets in order to watch for changes.
+
+When a referenced Secret is modified the operator will automatically perform a rolling restart of the Keycloak Deployment to pick up the changes.
```

Comment:
"...a referenced Secret is modified, the operator..."

Reply:
```suggestion
When a referenced Secret is modified, the operator automatically performs a rolling restart of the Keycloak Deployment to pick up the changes.
```

Reply:
(changes from bruno incorporated in suggestion)